Packs/EclecticIQ/Integrations/EclecticIQIntelligenceCenterv3/EclecticIQIntelligenceCenterv3.py

```diff
@@ -292,6 +292,16 @@ def send_api_request(self, method, path, params=None, data=None):
                     proxies=self.proxies,
                     timeout=30,
                 )
+            elif method == "patch":
+                r = requests.patch(
+                    url,
+                    headers=self.headers,
+                    params=params,
+                    data=json.dumps(data),
+                    verify=self.verify_ssl,
+                    proxies=self.proxies,
+                    timeout=30,
+                )
```

Comment:
The code is very long and really double
It seems advisable to make it more generic
something like that

```
def send_api_request(self, method, path, params=None, data=None):
    url = self.baseurl + path

    r = None
    try:
        if hasattr(requests, method):
            request_method = getattr(requests, method)
            r = request_method(
                url,
                headers=self.headers,
                params=params,
                data=json.dumps(data),
                verify=self.verify_ssl,
                proxies=self.proxies,
                timeout=30,
            )
        else:
            self.eiq_logging.error("Unknown method: " + str(method))
            raise Exception("Unsupported HTTP method: {}".format(method))
     

Continuation of the function
...

Reply:
@aaron1535 it looks a nice change, thanks for raising it, I have updated the MR with this fix